### PR TITLE
My Initial Commit

### PR DESCRIPTION
**Pull request by md3822**
Java Sample Application for Ticket 11371 with the following methods:
1. Equality match on a single nested field
2. Equality match on two nested fields
3. Use of an operator on a nested field
4. Equality match that uses a nested field and one or more fields that are not nested

### CODE REVIEW

File: EmbeddedDocumentsQuery/src/EmbeddedDocumentsQuery.java [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+			return queryOrders;
+		}
+		catch (Exception e) {
+			//log the exception
+			return null;
+		}		
+	}
+	
+	public List<Document> NestedNonNestedEqualityMatch(String connectionString, String city, Double total)
+	{
```
Comment (shinyou21) [inline EmbeddedDocumentsQuery/src/EmbeddedDocumentsQuery.java]: As shannon mentioned, the method names should reflect what the method does (example: filterByCity) rather than the operation it does

File: EmbeddedDocumentsQuery/testsrc/EmbeddedDocumentsQueryTest.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+	    //verify record
+	    Document result = filteredDocuments.get(0);
+	    String id = result.getString("_id");
+	    assertEquals("5a989e4abf545f22ed33d590", id);	    
+	}
+	
+	
+	@Test
+	public void TwoNestedFieldsEqualityMatchTest() {
+		EmbeddedDocumentsQuery embeddedDocumentsQuery = new EmbeddedDocumentsQuery();
```
Comment (shinyou21) [inline EmbeddedDocumentsQuery/testsrc/EmbeddedDocumentsQueryTest.java]: Needs a more descriptive name.

File: EmbeddedDocumentsQuery/testsrc/EmbeddedDocumentsQueryTest.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+import java.util.List;
+import org.bson.Document;
+import org.junit.Test;
+
+import com.mongodb.MongoClient;
+import com.mongodb.MongoClientURI;
+
+
+public class EmbeddedDocumentsQueryTest {
```
Comment (shinyou21) [inline EmbeddedDocumentsQuery/testsrc/EmbeddedDocumentsQueryTest.java]: Needs a more descriptive name.

File: EmbeddedDocumentsQuery/testsrc/EmbeddedDocumentsQueryTest.java [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+	public void TestConnection() {		
+		MongoClientURI clientUri = new MongoClientURI(CONNECTION_STRING);
+		try(MongoClient client = new MongoClient(clientUri))
+		{
+			assertNotNull(client);
+		}    
+	}
+
+	@Test
+	public void SingleNestedFieldEqualityMatchTest() {
```
Comment (shinyou21) [inline EmbeddedDocumentsQuery/testsrc/EmbeddedDocumentsQueryTest.java]: Needs a more descriptive name.

File: EmbeddedDocumentsQuery/testsrc/EmbeddedDocumentsQueryTest.java [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+	    
+	    //verify record
+	    Document result = filteredDocuments.get(0);
+	    String id = result.getString("_id");
+	    assertEquals("5a989e4a86d089e9b9c6ea65", id);
+	    
+	}
+	
+	@Test
+	public void NestedNonNestedEqualityMatchTest() {
```
Comment (shinyou21) [inline EmbeddedDocumentsQuery/testsrc/EmbeddedDocumentsQueryTest.java]: Needs a more descriptive name.

File: EmbeddedDocumentsQuery/src/EmbeddedDocumentsQuery.java [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+			
+			return queryOrders;
+		}
+		catch (Exception e) {
+			//log the exception
+			return null;
+		}		
+	}
+	
+	public List<Document> NestedNonNestedEqualityMatch(String connectionString, String city, Double total)
```
Comment (shinyou21) [inline EmbeddedDocumentsQuery/src/EmbeddedDocumentsQuery.java]: Needs a more descriptive name.

File: EmbeddedDocumentsQuery/src/EmbeddedDocumentsQuery.java [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+public class EmbeddedDocumentsQuery {
+	
+	/**
+	 * Retrieves documents filtered by shipping address - city.
+	 * @param connectionString: To MongoDB instance/MongoDB Cluster.
+	 * @param city: Shipping address - city to filter documents.
+	 * @return List<Document>: ArrayList of matching documents.
+	 */
+	
+	public List<Document> SingleNestedFieldEqualityMatch(String connectionString, String city){
```
Comment (shinyou21) [inline EmbeddedDocumentsQuery/src/EmbeddedDocumentsQuery.java]: As shannon mentioned, the method names should reflect what the method does (example: filterByCity) rather than the operation it does

File: EmbeddedDocumentsQuery/src/EmbeddedDocumentsQuery.java [hunk 8]
```diff
@@ -0,0 +1,127 @@
...
+	
+	/**
+	 * Retrieves documents filtered by shipping address - state and city.
+	 * @param connectionString: To MongoDB instance/MongoDB Cluster.
+	 * @param state: Shipping address - state to filter documents.
+	 * @param city: Shipping address - city to filter documents.
+	 * @return List<Document>: ArrayList of matching documents.
+	 */
+	
+	public List<Document> TwoNestedFieldsEqualityMatch(String connectionString, String state, String city){
```
Comment (shinyou21) [inline EmbeddedDocumentsQuery/src/EmbeddedDocumentsQuery.java]: need a more descriptive name

File: EmbeddedDocumentsQuery/src/EmbeddedDocumentsQuery.java (hunk 6 above)
Comment (shinyou21) [inline EmbeddedDocumentsQuery/src/EmbeddedDocumentsQuery.java]: need a more descriptive name

File: EmbeddedDocumentsQuery/src/EmbeddedDocumentsQuery.java (hunk 7 above)
Comment (shannonbradshaw) [reply EmbeddedDocumentsQuery/src/EmbeddedDocumentsQuery.java]: Yes. Please.

File: EmbeddedDocumentsQuery/src/EmbeddedDocumentsQuery.java (hunk 8 above)
Comment (shannonbradshaw) [reply EmbeddedDocumentsQuery/src/EmbeddedDocumentsQuery.java]: Please.

File: EmbeddedDocumentsQuery/src/EmbeddedDocumentsQuery.java (hunk 6 above)
Comment (shannonbradshaw) [reply EmbeddedDocumentsQuery/src/EmbeddedDocumentsQuery.java]: Also it's unlikely that we'd do an equality match on total. Let's change this to a query on postal code and sku in lineitems. This will illustrate the use of embedded documents and embedded documents in arrays.

File: EmbeddedDocumentsQuery/testsrc/EmbeddedDocumentsQueryTest.java (hunk 4 above)
Comment (shannonbradshaw) [inline EmbeddedDocumentsQuery/testsrc/EmbeddedDocumentsQueryTest.java]: This test needs to test more possible inputs such as an empty string, a value that results in no matches, a value that results in exactly one match, and a value that results in multiple documents matched. You need to test all cases so that we know your example works in the face of changes to the underlying dataset.

File: EmbeddedDocumentsQuery/testsrc/EmbeddedDocumentsQueryTest.java [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+	    
+	    //verify record
+	    Document result = filteredDocuments.get(0);
+	    String id = result.getString("_id");
+	    assertEquals("5a989e4abf545f22ed33d590", id);	    
+	}
+	
+	
+	@Test
+	public void TwoNestedFieldsEqualityMatchTest() {
```
Comment (shannonbradshaw) [inline EmbeddedDocumentsQuery/testsrc/EmbeddedDocumentsQueryTest.java]: This test needs to test more possible inputs such as an empty string, a value that results in no matches, a value that results in exactly one match, and a value that results in multiple documents matched. You need to test all cases so that we know your example works in the face of changes to the underlying dataset.

File: EmbeddedDocumentsQuery/testsrc/EmbeddedDocumentsQueryTest.java [hunk 10]
```diff
@@ -0,0 +1,84 @@
...
+	    
+	    //verify record
+	    Document result = filteredDocuments.get(0);
+	    String id = result.getString("_id");
+	    assertEquals("5a989e4a151dd8d28832eb2f", id);
+	    
+	}
+	
+	@Test
+	public void OperatorNestedFieldTest() {
```
Comment (shannonbradshaw) [inline EmbeddedDocumentsQuery/testsrc/EmbeddedDocumentsQueryTest.java]: Again, here, include tests for many inputs.

File: EmbeddedDocumentsQuery/testsrc/EmbeddedDocumentsQueryTest.java (hunk 5 above)
Comment (shannonbradshaw) [inline EmbeddedDocumentsQuery/testsrc/EmbeddedDocumentsQueryTest.java]: Again, here, include tests for many inputs.

**Review by Raghav2018 — APPROVED** [review]
Please make sure to check for compilation error before checking in the code.

**Review by Raghav2018 — CHANGES_REQUESTED** [review]
Please make sure your code has no compilation errors and all the unit tests run before commit.
